src/main/java/com/softserve/actent/model/entity/User.java

```diff
@@ -1,38 +1,97 @@
 package com.softserve.actent.model.entity;
 
+import com.softserve.actent.resources.StringConstants;
 import lombok.Data;
 import lombok.NoArgsConstructor;
 import lombok.NonNull;
-import lombok.RequiredArgsConstructor;
-import org.hibernate.annotations.GenericGenerator;
 
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.GeneratedValue;
-import javax.persistence.GenerationType;
-import javax.persistence.Id;
-import javax.persistence.Table;
+import javax.persistence.*;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.Date;
+import java.util.List;
 
 @Entity
 @Table(name = "users")
 @NoArgsConstructor
-@RequiredArgsConstructor
 @Data
 public class User {
     @Id
-//    @GenericGenerator(name = "idGenerator", strategy = "increment")
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
     @NonNull
-    @Column(name = "first_name")
+    @NotBlank(message = StringConstants.EMPTY_USER_FIRST_NAME)
+    @Column(nullable = false)
     private String firstName;
 
     @NonNull
-    @Column(name = "last_name")
+    @NotBlank(message = StringConstants.EMPTY_USER_LAST_NAME)
+    @Column(nullable = false)
     private String lastName;
 
-    // TODO: make unique
     @NonNull
+    @NotBlank(message = StringConstants.EMPTY_USER_LOGIN)
+    @Column(unique = true, nullable = false)
     private String login;
+
+    @NotNull
+    @NotBlank(message = StringConstants.EMPTY_USER_EMAIL)
+    @Column(unique = true, nullable = false)
+    private String email;
+
+    @NonNull
+    @NotBlank(message = StringConstants.EMPTY_USER_PASSWORD)
+    @Column(nullable = false)
+    private String password;
+
+    @NonNull
+    @NotNull(message = StringConstants.EMPTY_USER_BIRTH_DATE)
+    @Column(nullable = false)
+    private Date birth_date;
+
+    @NonNull
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "avatar_id")
+    private Image avatar;
+
+    @NonNull
+    @NotNull(message = StringConstants.EMPTY_USER_LOCATION)
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "location_id", nullable = false)
```

Comment:
Remove @JoinColumn annotation(we use mappedBy parameter).

Reply:
mappedBy parameter exist only in @OneToMany

Reply:
It is true, so you do not need anything here, only @ManyToOne, if on the other side we will have mappedBy parameter.